tests/adapters/switches/juniper_test.py

```diff
@@ -4312,11 +4312,11 @@ def test_get_bond(self):
 
         assert_that(if3.number, equal_to(3))
         assert_that(if3.link_speed, equal_to('1g'))
-        assert_that(if3.interface.name, equal_to("ae3"))
-        assert_that(if3.interface.port_mode, equal_to(TRUNK))
-        assert_that(if3.interface.access_vlan, equal_to(None))
-        assert_that(if3.interface.trunk_native_vlan, equal_to(2000))
-        assert_that(if3.interface.trunk_vlans, equal_to([999, 1000, 1001]))
+        assert_that(if3.name, equal_to('ae3'))
```

Comment:
If we do this we're still stuck with having the name of a bond in the contract.

i think baseinterface should not have the name, name should be on physical interface and bond should have a ._interface_name with the underscore  so that nobody wants to depend on that and its only use is to make the InterfaceObject when calling bond.interface